### PR TITLE
Add telemetry support

### PR DESCRIPTION
**Pull request by rajagopalans**

### CODE REVIEW

File: tests/test_telemetry.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      "hostname": "100.123.0.8",
+      "port": 64428,
+      "id": "d8f5a012-232c-4aca-b5c9-637dbba8cb4b"
+    }
+  ]
+}
+
+def test_add_endpoint():
+    mgr.add_endpoint("fakehost", "fakeport", "faketype")
+    mock_rest.json_resp_post.assert_called_with(uri='/api/streaming-config', data={'hostname': 'fakehost', 'port': 'fakeport', 'streaming_type': 'faketype', 'sequencing_mode': 'sequenced', 'protocol': 'protoBufOverTcp'})
```
Comment (that1guy15) [inline tests/test_telemetry.py:128]: wrap this line please. Suggest running black on all files you worked with.

**Review by that1guy15 — COMMENTED** [review]
Few NIT comments but overall look great. 
The one thing missing is a document in /docs/example-scripts with a walk-through of its usage. I usually use the same code from scripts and add some comentery around it. This is optional for this PR

File: aos/client.py [hunk 2]
```diff
@@ -4,13 +4,15 @@
...
+
+from aos.telemetry import AosTelemetryManager
 from .aos import AosRestAPI, AosAuth
 from .blueprint import AosBlueprint
 from .devices import AosDevices
 from .design import AosDesign
 from .resources import AosResources
 from .external_systems import AosExternalSystems
-
+from .telemetry import AosTelemetryManager
```
[the remark this answers is not in the document]
Comment (rajagopalans) [reply aos/client.py:15]: removed the one in line 8

File: aos/telemetry.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    connection_time:str
+    last_tx_time: str
+    epoch: str
+    connection_reset_count: int
+    dns_log: List[dict]
+    disconnection_time: str
+    
+    @classmethod
+    def from_json (cls, d:dict):
+        return AosTelemetryEndpointStatus( connected=d.get("connected"), connection_log=d.get("connectionLog"), connection_time=d.get("connectionTime"), last_tx_time=d.get("lastTransmitedTime"),epoch=d.get("epoch"),connection_reset_count=d.get("connectionResetCount"), dns_log=d.get("dnsLog"), disconnection_time=d.get("disconnectionTime"))
```
[the remark this answers is not in the document]
Comment (rajagopalans) [reply aos/telemetry.py:27]: done

File: aos/telemetry.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    host: str
+    port: int
+    streaming_type: str
+    protocol: str
+    sequencing_mode: str
+    ep_status: AosTelemetryEndpointStatus
+
+    @classmethod
+    def from_json(cls, d:dict):
+        return AosTelemetryEndpoint(id=d.get("id"), host=d.get("host"), port=d.get("port"), streaming_type=d.get("streaming_type"), protocol=d.get("protocol"),sequencing_mode=d.get("sequencing_mode"), ep_status=AosTelemetryEndpointStatus.from_json(d.get("status")))
```
[the remark this answers is not in the document]
Comment (rajagopalans) [reply aos/telemetry.py:44]: done

File: tests/test_telemetry.py (hunk 1 above)
Comment (that1guy15) [reply tests/test_telemetry.py:132]: please address this comment.

File: tests/test_telemetry.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+import pytest
+from unittest import mock
+from unittest.mock import call
+
+from aos.telemetry import AosTelemetryManager
+
+mock_rest = mock.Mock()
+mgr = AosTelemetryManager(mock_rest)
+
+mock_rest.json_resp_get.return_value = {
```
[the remark this answers is not in the document]
Comment (that1guy15) [reply tests/test_telemetry.py:17]: please address.

File: tests/test_telemetry.py [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+    mgr.delete_all_endpoints()
+    mock_rest.delete.assert_called()
+
+def test_get_endpoints(aos_api_version):
+    fixture_path = f"aos/{aos_api_version}/telemetry/endpoints.json"
+    mock_rest.json_resp_get.return_value = json.loads(read_fixture(fixture_path))
+    r = mgr.get_endpoints()
+    assert len(r) == 3
+    mock_rest.json_resp_get.assert_called()
+
```
Comment (that1guy15) [inline tests/test_telemetry.py:144]: tox -e py38 flake8 fails multiple lines here. Please address and make sure it passes. 
```
flake8 run-test: commands[0] | flake8 aos/ tests/
tests/test_telemetry.py:2:1: F401 'telnetlib.AO' imported but unused
tests/test_telemetry.py:6:1: F401 'unittest.mock.call' imported but unused
tests/test_telemetry.py:9:1: F401 'tests.util.make_session' imported but unused
tests/test_telemetry.py:12:1: E302 expected 2 blank lines, found 0
tests/test_telemetry.py:130:1: E302 expected 2 blank lines, found 1
tests/test_telemetry.py:132:86: E501 line too long (220 > 85 characters)
tests/test_telemetry.py:134:1: E302 expected 2 blank lines, found 1
tests/test_telemetry.py:138:1: E302 expected 2 blank lines, found 1
tests/test_telemetry.py:144:1: W391 blank line at end of file
ERROR: InvocationError for command /Users/boothr/apstra-api-python/.tox/flake8/bin/flake8 aos/ tests/ (exited with code 1)
```

Comment (that1guy15) [reply tests/test_telemetry.py:144]: Which has me curious why the GitHub actions to run these test don't fail. Ill dig into this.

**Review by that1guy15 — COMMENTED** [review]
Looks good. please address comments on test file.

**Review by that1guy15 — APPROVED** [review]
LGTM

**Pull request merged**